packages/oauth/src/install-provider.ts

```diff
@@ -350,30 +346,33 @@ export class InstallProvider {
         this.logger.debug('Skipped to proceed with the built-in redirection as beforeRedirection returned false');
         return;
       }
-      const state = await this.stateStore.generateStateParam(_installOptions, new Date());
-      const stateCookie: string = this.buildSetCookieHeaderForNewState(state);
-      if (res.getHeader('Set-Cookie')) {
-        // If the cookies already exist
-        const existingCookies = res.getHeader('Set-Cookie') || [];
-        const allCookies: string[] = [];
-        if (Array.isArray(existingCookies)) {
-          allCookies.push(...existingCookies);
-        } else if (typeof existingCookies === 'string') {
-          allCookies.push(existingCookies);
+      let state;
+      if (this.stateStore) {
```

Comment:
Can you use `this.stateVerification` flag instead and place L340 - L342 that you've deleted inside the if statement?
```suggestion
      if (this.stateVerification) {
```

Reply:
I did that initially but it makes TS sad due to `this.stateStore` being optional.
<img width="609" alt="Screenshot 2022-07-04 at 07 30 31" src="https://user-images.githubusercontent.com/681614/177094946-9a3addb2-d43d-42a3-a2ce-67fa923cc0e4.png">

I can do: 
* `if (this.stateStore)`
* `if(this.stateVerification && this.stateStore)
* `await this.stateStore?.generateStateParam` but this gets complicated as `state` then becomes optional

Reply:
Sounds good 👍  Done.